fbpic/boundaries/boundary_communicator.py

```diff
@@ -192,9 +192,11 @@ def __init__( self, Nz, zmin, zmax, Nr, rmax, Nm, dt,
             # to the simulation or in case a galilean frame is used.
             cells_per_step = 2.*c*dt/self.dz
             # Maximum number of timesteps before a particle can reach the end
-            # of the guard region including the maximum number of cells (+/-3)
-            # it can affect with a "cubic" particle shape_factor.
-            self.exchange_period = int( (self.n_guard-3)/cells_per_step )
+            # of the half of guard region including the maximum number of cells
+            # (+/-3) it can affect with a "cubic" particle shape_factor.
+            # (Particles are only allowed to reside in half of the guard
+            # region as this is the stencil reach of the current correction)
+            self.exchange_period = int(((self.n_guard/2)-3)/cells_per_step)
```

Comment:
Although this is not required to be included in this PR, I left it here so that we don't forget about it.

Reply:
With the newest change in nz_inject (+3 cells filled with plasma to the right), we need to make sure that "exchange_period" is small enough in the case of a boosted frame simulation with streaming plasma, such that particles can never be initialized in a region greater than n_guard/2. (this is not important for this PR, but only for the PR of the cross_deposition)

@RemiLehe When we implement the cross_deposition, i think that no particles are allowed to be in the outer half of the guard_region (assuming that the stencil of the cross_depositon is n_guard/2), right?

Reply:
OK, thank. I'll review the PR over the weekend, and I'll try to answer the above question.